src/Cake.Common/Tools/Cake/CakeRunner.cs

```diff
@@ -23,12 +23,34 @@ public sealed class CakeRunner : Tool<CakeSettings>
         private readonly ICakeEnvironment _environment;
         private readonly IFileSystem _fileSystem;
         private readonly IGlobber _globber;
-        private static readonly IEnumerable<FilePath> _executingAssemblyToolPaths = new FilePath[] { System.Reflection.Assembly.GetEntryAssembly().Location };
+        private static readonly IEnumerable<FilePath> _executingAssemblyToolPaths = GetEntryAssembly();
+
+        private static IEnumerable<FilePath> GetEntryAssembly()
+        {
+            string location = null;
+            try
+            {
+                location = System.Reflection.Assembly.GetEntryAssembly()?.Location;
+            }
+            catch
+            {
+            }
+            if (!string.IsNullOrWhiteSpace(location))
+            {
+                yield return location;
+            }
+            else
+            {
+                var directory = new FilePath(System.Reflection.IntrospectionExtensions.GetTypeInfo(typeof(CakeRunner)).Assembly.Location).GetDirectory();
+                yield return directory.CombineWithFilePath("Cake.exe");
+                yield return directory.CombineWithFilePath("Cake.dll");
+            }
+        }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="CakeRunner"/> class.
         /// </summary>
-        /// <param name="fileSystem">The file system.</param>
+        /// <param name="fileSystem">The file system.</param>11
```

Comment:
@devlead ha ha, thought it might be something like that. Lucky it wasn't a password :stuck_out_tongue: